ps_mainmenu.php

```diff
@@ -942,20 +945,37 @@ public function hookCategoryUpdate($params)
 
     protected function getCacheDirectory()
     {
-        return _PS_CACHE_DIR_ . 'ps_mainmenu';
+        $dir =_PS_CACHE_DIR_ . 'ps_mainmenu';
+
+        if (isset($this->context->customer)) {
+            $groups = $this->context->customer->getGroups();
+            if (count($groups)) {
+                $dir .=  '/' . implode('_', $groups);
+            }
+        }
+        if (!is_dir($dir)) {
+            mkdir($dir, 0775, true);
+        }
+
+        return $dir;
     }
 
     protected function clearMenuCache()
     {
-        $dir = $this->getCacheDirectory();
-
-        if (!is_dir($dir)) {
-            return;
-        }
+        $this->cleanMenuCacheDirectory(_PS_CACHE_DIR_ . 'ps_mainmenu');
+    }
 
+    private function cleanMenuCacheDirectory(string $dir)
```

Comment:
```php
private function cleanMenuCacheDirectory(string $dir)
```
should be
```php
private function cleanMenuCacheDirectory($dir)
```
Scalar type hints are available in PHP 7 only

Reply:
Done :)